torch/csrc/jit/runtime/operator.cpp

```diff
@@ -289,6 +339,18 @@ void registerOperator(Operator&& op) {
   getRegistry().registerOperator(std::move(op));
 }
 
+void deregisterOperator(const FunctionSchema& schema)
+{
+  // Do not attempt to de-register operator from the global shutdown
+  // As there are no guarantee about order of global destructors
```

Comment:
This is not true. Destructors of global objects are called in reverse order of how they were initialized.

Reply:
However containers like vector don't guarantee destructor order.

Reply:
@smessmer thanks for the suggestion. Fixed the issue by returning RAII object from `at::Dispatcher::addRegistrationListener()` and retaining it in `torch::jit::(anonymous namespace)::Register`